server/loggerUtils/logger.ts

```diff
@@ -0,0 +1,53 @@
+import winston from "winston";
+import dotenv from "dotenv";
+dotenv.config();
+
+// Configure logger
+const { combine, errors, timestamp, json, prettyPrint } = winston.format;
+const logLevels = {
+  fatal: 0,
+  error: 1,
+  warn: 2,
+  info: 3,
+  debug: 4,
+  trace: 5,
+};
+
+export const logger = winston.createLogger({
+  levels: logLevels,
+  level: process.env.NODE_ENV !== "dev" ? "info" : "trace",
+  format: combine(errors({ stack: true }), timestamp(), json(), prettyPrint()),

```

Comment:
The logs right now look like this:

```
{
message: 'GET /api/events/view-event-special-groups/?eventIds=recDf5ApLKnUEFH37,recCRsBd2GOZjeWlm,recTmkibqvV3jZdXo',
level: 'info',
timestamp: '2023-02-23T00:51:18.223Z'
}
{
message: 'GET /api/neighborhoods/?neighborhoodIds=rec2yhFgOpV8jbftP,recH9I6SqH1bJLOWy,recLkrByIe8Qf0TO6,recAPkAEcQozCESRr,rec1p9vh8UV9OjO1U,recLkrByIe8Qf0TO6,recqqrcNqyQJmownU,recAPkAEcQozCESRr,recqqrcNqyQJmownU,recAPkAEcQozCESRr,rec2yhFgOpV8jbftP,recH9I6SqH1bJLOWy,recLkrByIe8Qf0TO6,recjhBWtxACnXUDZO,reciLwk1Q7prqFbKJ,recbXVC3wCuQlDg49,reckcpfsL91JTlpx9,rec1p9vh8UV9OjO1U,recISBbWGW6ldlV5i,rec2yhFgOpV8jbftP,recH9I6SqH1bJLOWy,recLkrByIe8Qf0TO6,reckcpfsL91JTlpx9,rec1p9vh8UV9OjO1U',
level: 'info',
timestamp: '2023-02-23T00:58:08.886Z'
}
{
message: 'GET /api/volunteers/drivers',
level: 'info',
timestamp: '2023-02-23T00:58:11.180Z'
}
```
This multi-line JSON is overly spread out for human reading and ability to grep. For server logs, I much prefer one-line logs per event and timestamp at the beginning. Something like this:

```
[2023-02-23T00:51:18.223Z] INFO: GET /api/events/view-event-special-groups/?eventIds=recDf5ApLKnUEFH37,recCRsBd2GOZjeWlm
```
For our use case, we should optimize for human readability.  JSON format (still in a single line) is better if it was important to have machine-readable logs for ingesting into something like Splunk, but that's beyond the scope and needs of this project. We'll be looking at logs through the Railway webpage and using its built-in filter.
If Winston has some built-in human-optimized format, let's try that. single-line per event and timestamp at the front is a must though.

Reply:
Got it. Thanks for helping me through this. Winston has the ability to define custom formats. The format I've defined matches the one above with the addition of a stack trace for throws at the end of the line. Below is an example with a test error I threw for to test the format. Is this on the right track?

`[2023-02-23T04:05:49.609Z] INFO: Incorrect password 
[2023-02-23T04:05:54.541Z] ERROR: User doesn't exist STACK: Error: User doesn't exist
     at C:\development\grassroots-admin\server\routes\auth.ts:124:20
     at step (C:\development\grassroots-admin\server\routes\auth.ts:33:23)
     at Object.next (C:\development\grassroots-admin\server\routes\auth.ts:14:53)
     at fulfilled (C:\development\grassroots-admin\server\routes\auth.ts:5:58)
     at processTicksAndRejections (node:internal/process/task_queues:95:5)`